File: src/main/java/com/leet/medium/ZigZagConversion.ts
export function convert(s: string, numRows: number): string {
	if (numRows <= 1) {
			return s;
	}
	const ans = new Array(numRows).fill('');

	let row = 0;
	let down = true;
	for (let i = 0; i < s.length; i++) {
			ans[row] += s[i];
			if (row === 0) {
					down = true;
			} else if (row === (numRows-1)) {
					down = false;
			}
			row += down ? 1 : -1;
	}
	return ans.join('');
}

File: src/main/java/com/leet/medium/ZigZagConversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if numRows == 1:
            return s

        ans = [[] for i in range(0, numRows)]
        row, down = 0, True

        for i in range(0, len(s)):
            ch = s[i]
            ans[row].append(ch)

            if row == (numRows-1) and i != 0:
                down = False
            elif row == 0:
                down = True

            if down:
                row += 1
            else:
                row -= 1

        return ''.join([''.join(row) for row in ans])
